modules/aggregation-layers/src/utils/gpu-grid-aggregation/gpu-grid-aggregator.js

```diff
@@ -341,7 +341,10 @@ export default class GPUGridAggregator {
   validateProps(aggregationParams, opts) {
     const {changeFlags, projectPoints, gridTransformMatrix} = aggregationParams;
     log.assert(
-      changeFlags.dataChanged || changeFlags.viewportChanged || changeFlags.cellSizeChanged
+      changeFlags.dataChanged ||
```

Comment:
This feature (generating smooth edges) should only operate on data produced by aggregation, you shouldn't be doing any changes to this file. Any reason you have to make these changes? If not needed remove.

Reply:
I added this change to handle Linear Interpolation checkbox in the example, after changing the value for linear interpolation the contour layer was not refreshing, so I added this flag to handle this.

When we will remove the Linear Interpolation checkbox we will remove this change, It just a temporary change.

Reply:
What you really need is check for this flag in `_shouldRebuildContours`, and when linear interpolation mode changed, this method should return true and it will re trigger contour generation.